### PR TITLE
Issue #11277: update code base to have javadoc tag to explain noinspection content

### PR DESCRIPTION
**Pull request by nick-mancuso**
Related to #11277, takes care of HtmlTagCanBeJavadocTag, JavadocReference, BooleanParameter, MethodParameterNamingConvention

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java [hunk 1]
```diff
@@ -699,7 +699,7 @@ private static boolean isSetterMethod(DetailAST ast) {
         private static String getJavaDocText(DetailAST node) {
             final String text = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<document>\n"
                     + node.getFirstChild().getText().replaceAll("(^|\\r?\\n)\\s*\\* ?", "\n")
-                            .replaceAll("\\n@noinspection.*\\r?\\n", "\n")
+                            .replaceAll("\\n?@noinspection.*\\r?\\n[^@]*", "\n")
```
Comment (nick-mancuso) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java:702]: Required to not add `noinspectionreason` content to xdocs

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/StringLiteralEqualityCheck.java [hunk 2]
```diff
@@ -84,6 +84,8 @@
  *
  * @since 3.2
  * @noinspection HtmlTagCanBeJavadocTag
+ * @noinspectionreason HtmlTagCanBeJavadocTag - encoded symbols were not decoded
+ *      when replaced with Javadoc tag
```
Comment (nick-mancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/StringLiteralEqualityCheck.java:88]: This was the case that caused the need for https://github.com/checkstyle/checkstyle/pull/11880/files#r917429288

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java (hunk 1 above)
Comment (nick-mancuso) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java:702]: Build will fail without this change: https://github.com/checkstyle/checkstyle/pull/11880/checks?check_run_id=7272320294

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java [hunk 3]
```diff
@@ -456,6 +456,7 @@ private void storePrevScopeUninitializedVariableData() {
      * @param prevScopeUninitializedVariableData variable for previous stack of uninitialized
      *     variables
      * @noinspection MethodParameterNamingConvention
+     * @noinspectionreason complicated check requires descriptive naming
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java:459]: MethodParameterNamingConvention 
name is missed

**Review by romani — CHANGES_REQUESTED** [review]
items:

**Review by rnveach — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java (hunk 3 above)
Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java]: Thanks, done.

**Pull request merged**